cmake fails to find `<gio/gunixmounts.h>` in `collect.c` on NixOS for version 3.8

**Did you buy darktable from an application store ?** No.

**Describe the bug/issue**


**To Reproduce**
When I try to compile, I get the error:

```
FAILED: nix/store/4xnai9jci10xlxmw7hf0p6nrg3mnhlxf-darktable-3.8.0/lib/darktable/plugins/lighttable/CMakeFiles/collect.dir/collect.c.o 
/nix/store/da9l6ifizmh883j94gzx5n51h8lrrp00-gcc-wrapper-10.3.0/bin/gcc -DAVIF_DLL -DDT_HAVE_SIGNAL_TRACE -DGDK_DISABLE_DEPRECATED -DGDK_VERSION_MIN_REQUIRED=GDK_VERSION_3_22 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_MIN_REQUIRED -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_56 -DGTK_DISABLE_DEPRECATED -DGTK_DISABLE_SINGLE_INCLUDES -DHAVE_BUILTIN_CPU_SUPPORTS -DHAVE_CONFIG_H -DHAVE_GAME -DHAVE_GMIC -DHAVE_GPHOTO2 -DHAVE_GPHOTO_25_OR_NEWER -DHAVE_GRAPHICSMAGICK -DHAVE_HTTP_SERVER -DHAVE_ICU -DHAVE_ISO_CODES -DHAVE_KWALLET -DHAVE_LENSFUN -DHAVE_LIBAVIF=1 -DHAVE_LIBHEIF=1 -DHAVE_LIBRAW=1 -DHAVE_LIBSECRET -DHAVE_MAP -DHAVE_OPENCL -DHAVE_OPENEXR -DHAVE_OPENJPEG -DHAVE_OSMGPSMAP_110_OR_NEWER -DHAVE_OSMGPSMAP_NEWER_THAN_110 -DHAVE_PRINT -DHAVE_SQLITE_324_OR_NEWER -DNATIVE_ARCH -DSQLITE_CORE -DSQLITE_ENABLE_ICU -DUSE_COLORDGTK -DUSE_LUA -D_RELEASE -D_XOPEN_SOURCE=700 -D__GDK_KEYSYMS_COMPAT_H__ -Dcollect_EXPORTS -I/build/darktable-3.8.0/build/nix/store/4xnai9jci10xlxmw7hf0p6nrg3mnhlxf-darktable-3.8.0/lib/darktable/plugins/lighttable -I/build/darktable-3.8.0/src/libs -I/build/darktable-3.8.0/src -I/build/darktable-3.8.0/src/external/LuaAutoC -I/build/darktable-3.8.0/build/nix/store/4xnai9jci10xlxmw7hf0p6nrg3mnhlxf-darktable-3.8.0/bin -isystem /build/darktable-3.8.0/src/external -isystem /build/darktable-3.8.0/src/external/OpenCL -isystem /nix/store/5idik4vmd83cnx5z4shg44s53l4n2whj-glib-2.70.1-dev/include/glib-2.0 -isystem /nix/store/mbgcj6maiy41m0hacf1pw5nw56m3wn3q-glib-2.70.1/lib/glib-2.0/include -isystem /nix/store/wqqwjjyygmjgmbmkhncq55yzrfgsj1ks-cairo-1.16.0-dev/include/cairo -isystem /nix/store/srqsdxbv4nsxmyl2nl1cpjwbxhawvzxg-gtk+3-3.24.30-dev/include/gtk-3.0 -isystem /nix/store/1y14888d7hg8djh0dzihlmj6905wrlmz-atk-2.36.0-dev/include/atk-1.0 -isystem /nix/store/kic3nfhbd52fnkvv8ymz14c7rx2x79mf-gdk-pixbuf-2.42.6-dev/include/gdk-pixbuf-2.0 -isystem /nix/store/xhfhdv452b3c9c8fvg160i6ag29kkzf4-pango-1.48.10-dev/include/pango-1.0 -isystem /nix/store/wp2lfivjbh8jjz09c5wghl40b2qjqfb5-harfbuzz-3.0.0-dev/include/harfbuzz -isystem /nix/store/jaw6h3zvnwlyl1jajdi0nbalwyq9f0v6-libxml2-2.9.12-dev/include/libxml2 -isystem /nix/store/7fb499a00xjdwazahl4m7s3w2gm2ch3c-libsoup-2.74.1-dev/include/libsoup-2.4 -isystem /nix/store/70j03g5609ksfiasylh7cgvplz9j2xh0-lensfun-0.3.2/include/lensfun -isystem /nix/store/8fx401pg1kb0l97g0517dybn9lcvvgfa-librsvg-2.52.3-dev/include/librsvg-2.0 -isystem /nix/store/4zb8wy4dp414vczn7yxxvs84pdj3rmx0-json-glib-1.6.6-dev/include/json-glib-1.0 -isystem /nix/store/3zby8md8vzkhss3j8m4916gif98q8q47-openjpeg-2.4.0-dev/include/openjpeg-2.4 -isystem /nix/store/qd1xyrbclka0lx1799lfxg2aylwmdpq4-libsecret-0.20.4-dev/include/libsecret-1 -isystem /nix/store/4qmsa0m2dqlipmxz9z1vr5lnwmxak0w4-graphicsmagick-1.3.36/include/GraphicsMagick -isystem /nix/store/pyz5dck6i6nn3rs3wvbf53hsfq38ygmz-osm-gps-map-1.2.0-dev/include/osmgpsmap-1.0 -isystem /nix/store/84rcvs6s32d217jgr6s16zsrf450qw0v-colord-1.4.5-dev/include/colord-1 -isystem /nix/store/4d2hmgvwxx5zn9qv9zkpddcf9l9sqjya-colord-gtk-0.2.0-dev/include/colord-1 -isystem /nix/store/hxf9dx7n8ii88bb00gvla0zncg4xgj4b-openexr-3.1.2-dev/include/OpenEXR -isystem /nix/store/j741vsis2kfcfrvf8y4s15s36hrqdm2l-imath-3.1.3/include/Imath -Wall -Wformat -Wformat-security -Wshadow -Wtype-limits -Wvla -Wold-style-declaration -Wno-unknown-pragmas -Wno-error=varargs -Wno-format-truncation -Wno-error=address-of-packed-member -std=c99 -fopenmp -march=native -msse2 -g -mfpmath=sse -O3 -DNDEBUG -O3 -ffast-math -fno-finite-math-only -fexpensive-optimizations -fPIC -fvisibility=hidden   -include common/module_api.h -include libs/lib_api.h -MD -MT nix/store/4xnai9jci10xlxmw7hf0p6nrg3mnhlxf-darktable-3.8.0/lib/darktable/plugins/lighttable/CMakeFiles/collect.dir/collect.c.o -MF nix/store/4xnai9jci10xlxmw7hf0p6nrg3mnhlxf-darktable-3.8.0/lib/darktable/plugins/lighttable/CMakeFiles/collect.dir/collect.c.o.d -o nix/store/4xnai9jci10xlxmw7hf0p6nrg3mnhlxf-darktable-3.8.0/lib/darktable/plugins/lighttable/CMakeFiles/collect.dir/collect.c.o -c /build/darktable-3.8.0/src/libs/collect.c
/build/darktable-3.8.0/src/libs/collect.c:39:10: fatal error: gio/gunixmounts.h: No such file or directory
   39 | #include <gio/gunixmounts.h>
      |          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
```

**Platform**
_Please fill as much information as possible in the list given below. Please state "unknown" where you do not know the answer and remove any sections that are not applicable _

* darktable version : e.g. 3.8.0
* OS                : NixOS 21.11 / kernel 5.10.81
* Linux - Distro    : NixOS 21.11
* Memory            : 
* Graphics card     : 
* Graphics driver   : 
* OpenCL installed  : no
* OpenCL activated  : no
* Xorg              : Yes
* Desktop           : Plasma
* GTK+              : 
* gcc               : 
* cflags            : 
* CMAKE_BUILD_TYPE  : release

**Additional context**

I've tried a few variations on the patch https://github.com/paperdigits/darktable/commit/9a8cb704da40417a8f321ecc25f4c0dc95aa68d3.patch but none of them work.

--- a/cmake/modules/FindGIO.cmake
+++ b/cmake/modules/FindGIO.cmake
@@ -11,7 +11,7 @@ include(LibFindMacros)
 libfind_package(GIO Glib)
 
 # Use pkg-config to get hints about paths
-libfind_pkg_check_modules(GIO_PKGCONF gio-2.0)
+libfind_pkg_check_modules(GIO_PKGCONF gio-2.0 gio-unix-2.0)
 
 # Find the library
 find_library(GIO_LIBRARY
@@ -22,4 +22,6 @@ find_library(GIO_LIBRARY
 # Set the include dir variables and the libraries and let libfind_process do the rest.
 # NOTE: Singular variables for this library, plural for libraries this lib depends on.
 libfind_process(GIO)
+
+list(APPEND GIO_INCLUDE_DIRS ${GIO_PKGCONF_INCLUDE_DIRS})
 

